### PR TITLE
Bug 1698800 - Conform datetime metric reference page to new format

### PR DESCRIPTION
**Pull request by brizental**
...and Add Glean.js information to datetime metric page.

I also found a bug in the Swift example and fixed it (thanks @travis79 !).

[Rendered](https://innate-earth.surge.sh/reference/metrics/datetime.html).

Somethings I noticed:

- Java and Kotlin don't have the `testGetValue` API documented, was that on purpose? [That API seems to exist](https://github.com/mozilla/glean/blob/main/glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt#L185). I can just include the missing examples in this PR, but wanted to double check first.
- Firefox Desktop `testGetValue` actually works like the `testGetValueAsString`. Worth filing a bug or was this on purpose?

### CODE REVIEW

**Review by mdboom — APPROVED** [review]
This looks great.  It seems like the missing languages for testGetValue is just an oversight so we should put that in.

As for the Firefox Desktop question, I defer to @chutten .

**Comment by chutten** [comment]
>     * Firefox Desktop `testGetValue` actually works like the `testGetValueAsString`. Worth filing a bug or was this on purpose?

Just because it was on purpose doesn't mean it doesn't need a bug : )

Hm, but to know for certain whether or not this was intentional we must ask the person responsible for implementing the API. Now [who was that](https://bugzilla.mozilla.org/show_bug.cgi?id=1673643)....  : D

At the very least we should consider this approach, and a bug could serve as documentation for why we choose to do it or not. I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1709147 for that purpose. Thank you for bringing this up!

File: docs/user/reference/metrics/datetime.md [hunk 1]
```diff
@@ -124,149 +58,362 @@ import datetime
...
+
+<div data-lang="Firefox Desktop" class="tab"></div>
+
 {{#include ../../../shared/tab_footer.md}}
 
-## Limits
+### `testGetNumRecordedErrors`
 
-* None.
+Gets number of errors recorded for a given counter metric.
```
Comment (badboy) [inline docs/user/reference/metrics/datetime.md:310]:
```suggestion
Get number of errors recorded for a given datetime metric.
```

File: docs/user/reference/metrics/datetime.md [hunk 2]
```diff
@@ -124,149 +58,362 @@ import datetime
...
 ```
 
 </div>
+{{#include ../../../shared/tab_footer.md}}
 
-<div data-lang="JS" class="tab">
+### `testGetValueAsString`
 
-> **Note**: JS APIs are only available in Firefox Desktop.
+Gets the recorded value for a given datetime metric an [ISO Date String](https://en.wikipedia.org/wiki/ISO_8601#Dates).
```
Comment (badboy) [inline docs/user/reference/metrics/datetime.md:191]:
```suggestion
Get the recorded value for a given datetime metric as an [ISO Date String](https://en.wikipedia.org/wiki/ISO_8601#Dates).
```

File: docs/user/reference/metrics/datetime.md [hunk 3]
```diff
@@ -124,149 +58,362 @@ import datetime
...
-// Records "now"
-install::first_run.set(None);
-// Records a custom datetime
-let custom_date = FixedOffset::east(0).ymd(2019, 3, 25).and_hms(0, 0, 0);
-install::first_run.set(Some(custom_date));
+## Testing API
+
+### `testGetValue`
+
+Gets the recorded value for a given datetime metric as a language specific Date object.
```
Comment (badboy) [inline docs/user/reference/metrics/datetime.md:121]:
```suggestion
Get the recorded value for a given datetime metric as a language-specific Date object.
```

**Review by badboy — CHANGES_REQUESTED** [review]

File: docs/user/reference/metrics/datetime.md [hunk 4]
```diff
@@ -124,149 +58,362 @@ import datetime
...
+Gets number of errors recorded for a given counter metric.
 
-## Examples
+{{#include ../../../shared/tab_header.md}}
 
-* When did the user first run the application?
+<div data-lang="Kotlin" class="tab">
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Install
```
Comment (badboy) [inline docs/user/reference/metrics/datetime.md:317]:
```suggestion
import mozilla.telemetry.glean.testing.ErrorType
import org.mozilla.yourApplication.GleanMetrics.Install
```

File: docs/user/reference/metrics/datetime.md [hunk 5]
```diff
@@ -124,149 +58,362 @@ import datetime
...
+assertEquals(0, Install.firstRun.testGetNumRecordedErrors(ErrorType.InvalidValue))
+```
 
-* `invalid_value`: Setting the date time to an invalid value.
+</div>
+
+<div data-lang="Java" class="tab">
+
+```Java
+import org.mozilla.yourApplication.GleanMetrics.Install;
```
Comment (badboy) [inline docs/user/reference/metrics/datetime.md:327]:
```suggestion
import mozilla.telemetry.glean.testing.ErrorType
import org.mozilla.yourApplication.GleanMetrics.Install;
```

**Pull request merged**